request-manager/jl/manager.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2017, Square, Inc.
+
+// Package jl provides and interface for managing Job Logs (JLs).
+package jl
+
+import (
+	"database/sql"
+
+	"github.com/square/spincycle/proto"
+	"github.com/square/spincycle/request-manager/db"
+)
+
+// A Manager is used to create and manage JLs.
+type Manager interface {
+	// Create saves a JL to the db.
+	Create(requestId string, jl proto.JobLog) (proto.JobLog, error)
+
+	// Get gets a single JL.
+	Get(requestId string, jobId string) (proto.JobLog, error)
+
+	// GetFull gets all of the JLs for a request.
+	GetFull(requestId string) ([]proto.JobLog, error)
+}
+
+type manager struct {
+	dbc db.Connector
+}
+
+func NewManager(dbc db.Connector) Manager {
+	return &manager{
+		dbc: dbc,
+	}
+}
+
+func (m *manager) Create(requestId string, jl proto.JobLog) (proto.JobLog, error) {
+	jl.RequestId = requestId
+
+	conn, err := m.dbc.Connect() // connection is from a pool. do not close
+	if err != nil {
+		return jl, err
+	}
+
+	q := "INSERT INTO job_log (request_id, job_id, try, type, started_at, finished_at, state, `exit`, " +
+		"error, stdout, stderr) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+	_, err = conn.Exec(q,
+		&jl.RequestId,
+		&jl.JobId,
+		&jl.Try,
+		&jl.Type,
+		&jl.StartedAt,
+		&jl.FinishedAt,
+		&jl.State,
+		&jl.Exit,
+		&jl.Error,
+		&jl.Stdout,
+		&jl.Stderr,
+	)
+	if err != nil {
+		return jl, err
+	}
+
+	// Update the finished job count on the request if the job completed successfully.
+	if jl.State == proto.STATE_COMPLETE {
+		q = "UPDATE requests SET finished_jobs = finished_jobs + 1 WHERE request_id = ?"
```

Comment:
If job-log component is only responsible for job logs, should it deal with/affect requests directly?